### PR TITLE
Whinier transitions

### PR DESCRIPTION
**Pull request by chriswoodrich**
I had some trouble tracking down some ```InvalidTransition``` errors, due to multiple callbacks on some of our aasm-included models.  This adds a little helpful info to the error message.  ```Event 'graduate' cannot transition from 'undergraduate'. Failed callback(s) [:paid_tuition?]```

If you all like where this is going, I'll add specs and clean up a bit.

### CODE REVIEW

**Comment by alto** [comment]
@chriswoodrich I like the idea. Keep on going! :+1:

**Comment by chriswoodrich** [comment]
@alovak will do!

**Comment by chriswoodrich** [comment]
@alovak have a look when you have a minute.  I did some abnormal stuff with testing to try to properly check for the new messages.

**Comment by alto** [comment]
@chriswoodrich Do you mean @alto ?

**Comment by chriswoodrich** [comment]
Heh. Yes.

**Comment by chriswoodrich** [comment]
@alto please have a look.  I'd appreciate any comments.

**Comment by chriswoodrich** [comment]
@alto any thoughts on this?

**Comment by alto** [comment]
@chriswoodrich Sorry, had some busy times recently. Your changes look great so far. Unfortunately, there were a couple of minor code changes which led to some conflicts with your PR. Do you mind trying to merge and resolve those? Will happily merge after (and do a release).

File: lib/aasm/errors.rb [hunk 1]
```diff
@@ -3,14 +3,18 @@ module AASM
...
+      @object, @event_name, @state_machine_name, @failures = object, event_name, state_machine_name, failures
     end
 
     def message
-      "Event '#{event_name}' cannot transition from '#{object.aasm(state_machine_name).current_state}'"
+      "Event '#{event_name}' cannot transition from '#{object.aasm(state_machine_name).current_state}'#{reasoning}"
+    end
+
+    def reasoning
+      ". Failed callback(s): #{@failures}." unless failures.empty?
```
Comment (alto) [inline lib/aasm/errors.rb]: Maybe the '.' should go into `message`. What do you think?

File: spec/unit/event_spec.rb [hunk 2]
```diff
@@ -109,6 +109,35 @@
     expect(event.fire(obj, {}, nil, 'arg1', 'arg2')).to eq(:closed)
   end
 
+  context 'when given a gaurd proc' do
+    it 'should have access to callback failures in the transitions' do
+      event = AASM::Core::Event.new(:graduate, state_machine) do
+        transitions :to => :alumni, :from => [:student, :applicant],
+          :guard => Proc.new { 1 + 1 == 3 }
+      end
+      line_number = __LINE__ - 2
```
Comment (alto) [inline spec/unit/event_spec.rb]: Nice :+1:

**Comment by chriswoodrich** [comment]
@alto I merged master and incorporated your changes re the error message.  Just waiting on rbx-2.5.8 to pass Travis.

**Comment by alto** [comment]
@chriswoodrich Great work, thanks a lot! Will release with the next version.

**Comment by alto** [comment]
I just released version `4.8.0` including the change.

**Pull request merged**